raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3302,19 +3297,34 @@ private long append(int epoch, List<T> records, OptionalLong requiredBaseOffset,
 
         BatchAccumulator<T> accumulator = leaderState.accumulator();
         boolean isFirstAppend = accumulator.isEmpty();
-        final long offset = accumulator.append(epoch, records, requiredBaseOffset, isAtomic);
+        final long offset = accumulator.append(epoch, records, true);
```

Comment:
Just curious, did we ever do non-atomic appends in the Raft layer before? I think the controller only did atomic appends (even though the controller itself would split up non-atomic batches)

Reply:
The controller used to use both atomic and non-atomic append. That logic was changed to do the record splitting in the metadata module and only use atomic appends. This was required so that the in-memory snapshots generated by the controller were not split across batches.

If the records in an in-memory snapshot were split across batch boundaries then the controller was not able to revert to an in-memory snapshot.

Unrelated to the original motivation of this PR but I took the opportunity to remove API/methods that were not being used.